ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -37,7 +38,21 @@ def authenticate(self, request, username=None, password=None, **kwargs):
         # This auth class doesn't create any new local users, so we just need to make sure
         # it has an AuthenticatorUser associated with it.
         if user:
-            AuthenticatorUser.objects.get_or_create(uid=username, user=user, provider=self.database_instance)
+            logger.info(f"user 2: {user.__dict__}")
+            user_attrs = {
+                "username": username,
+                "first_name": user.first_name,
+                "last_name": user.last_name,
+                "email": user.email,
+                "is_superuser": user.is_superuser,
+                "auth_time": now().isoformat().replace('+00:00', 'Z')
```

Comment:
This does raise eyebrows. What are you doing and why? I think you want to save the datetime in the JSON content in the same way it is presented in the API. What is the existing logic that the API uses to do this? I think the origin comes to this:

https://github.com/encode/django-rest-framework/blob/552a67acde5eb13f84ec9dbdf4363313cdfda2f0/rest_framework/fields.py#L1212-L1214

So my code preference would be to be deferential to the source logic. So instead of

```python
In [4]: now().isoformat().replace('+00:00', 'Z')
Out[4]: '2024-01-23T16:39:16.693724Z'
```

I would find this preferable

```python
In [7]: DateTimeField().to_representation(now())
Out[7]: '2024-01-23T16:44:02.038764Z'
```